lib/enigma.rb

```diff
@@ -39,9 +37,34 @@ def generate_shifts(key_hash, date_hash)
     key_hash.merge!(date_hash) { |k, o, n| o + n }
   end
 
-end
+  def find_index(character)
+     @character_set.index(character)
+  end
 
+  def shift_character(shift_number, character)
+    character_index = find_index(character)
+      if character_index
+        new_index = character_index + shift_number
+        @character_set[(new_index % 27)]
+      else
+        character
+      end
+  end
 
-#notes for whiteboarding
-#must require 'date'
-#@date = Date.today.strftime("%d/%m/%Y") require ‘date’
+  def shift_message(message, shifts)
+    split_message = message.downcase.split("")
```

Comment:
this method will handle the downcase and the index method will handle special character